dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/HvdcPV/HvdcPVDiagramPQ.mo

```diff
@@ -0,0 +1,124 @@
+within Dynawo.Electrical.HVDC.HvdcPV;
+
+/*
+* Copyright (c) 2015-2020, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+model HvdcPVDiagramPQ "Model of PV HVDC link with a PQ diagram. Each terminal can regulate the voltage or the reactive power, depending on the user's choice."
+  import Modelica;
+  import Dynawo.Electrical.HVDC;
+
+  extends HVDC.BaseClasses.BaseHvdcPDiagramPQ;
+  extends AdditionalIcons.Hvdc;
+
+/*
+  Equivalent circuit and conventions:
+
+               I1                  I2
+   (terminal1) -->-------HVDC-------<-- (terminal2)
+
+*/
+
+  Connectors.ZPin U1RefPu (value (start = ComplexMath.'abs'(u10Pu))) "Voltage regulation set point in p.u (base UNom) at terminal 1";
+  Connectors.ZPin U2RefPu (value (start = ComplexMath.'abs'(u20Pu))) "Voltage regulation set point in p.u (base UNom) at terminal 2";
+  Connectors.ZPin Q1RefPu (value (start = s10Pu.im)) "Reactive power regulation set point in p.u (base SnRef) (receptor convention) at terminal 1";
+  Connectors.ZPin Q2RefPu (value (start = s20Pu.im)) "Reactive power regulation set point in p.u (base SnRef) (receptor convention) at terminal 2";
+  Connectors.BPin modeU1 (value (start = modeU10)) "Boolean assessing the mode of the control: true if U mode, false if Q mode";
+  Connectors.BPin modeU2 (value (start = modeU20)) "Boolean assessing the mode of the control: true if U mode, false if Q mode";
+
+  parameter Boolean modeU10 "Start value of the boolean assessing the mode of the control at terminal 1: true if U mode, false if Q mode";
+  parameter Boolean modeU20 "Start value of the boolean assessing the mode of the control at terminal 2: true if U mode, false if Q mode";
+
+  type QStatus = enumeration (Standard "Reactive power is fixed to its initial value",
+                              AbsorptionMax "Reactive power is fixed to its absorption limit",
+                              GenerationMax "Reactive power is fixed to its generation limit");
+
+  output Types.Angle Theta1(start = UPhase10) "Angle of the voltage at terminal 1 in rad";
+  output Types.Angle Theta2(start = UPhase20) "Angle of the voltage at terminal 2 in rad";
```

Comment:
I don't really like to have prefix input/output in our models because it is very related to causal modeling, which is not what we do.

Reply:
Ok, I can remove the output